product_pricelist_direct_print/wizards/product_pricelist_print.py

```diff
@@ -218,3 +220,51 @@ def get_pricelist_to_print(self):
         if not pricelist and self.partner_count == 1:
             pricelist = self.partner_ids[0].property_product_pricelist
         return pricelist
+
+    def get_products_domain(self):
+        domain = [
+            ('sale_ok', '=', True)
+        ]
+        if self.categ_ids:
+            domain.append(('categ_id', 'in', self.categ_ids.ids))
+        return domain
+
+    def get_products_to_print(self):
+        self.ensure_one()
+        if self.last_ordered_products:
+            products = self.get_last_ordered_products_to_print()
+        else:
+            if self.show_variants:
+                products = self.product_ids or self.product_tmpl_ids.mapped(
+                    'product_variant_ids')
+            else:
+                products = self.product_tmpl_ids
+        if not products:
+            products = products.search(self.get_products_domain())
+        return products
+
+    def get_group_key(self, product):
+        return product.categ_id.name
+
+    def get_sorted_products(self, products):
+        if self.order_field == 'name':
+            products = products.sorted(lambda x: x.name)
+        elif self.order_field == 'default_code':
+            products = products.sorted(
+                lambda x: x.default_code or '')
+        return products
```

Comment:
Maybe something more generic?
```suggestion
    if self.order_field:
        # Using "or ''" to avoid issues with None type
        return products.sorted(lambda x: getattr(x, self.order_field) or '')
    return products
```

Reply:
:+1: changes done.